packit/config/aliases.py

```diff
@@ -34,143 +35,181 @@
+    for target in list(targets) or [default] if default else []:
+        if target in aliases:
+            # expand alias
+            result.update(aliases[target])
+        else:
+            result.update(
+                # try to convert target to Distro
+                [
+                    distro
+                    for distro in chain(aliases["fedora-all"], aliases["epel-all"])
+                    if target in distro
+                ]
+                # use the original string
+                or [target],
+            )
+    return result
 
 
 def get_branches(
-    *name: str,
-    default: str = DEFAULT_VERSION,
+    *targets: str,
+    default: Optional[str] = DEFAULT_VERSION,
     default_dg_branch: str = "main",
     with_aliases: bool = False,
 ) -> set[str]:
     """
-    Expand the aliases to the name(s) and transfer to the dist-git branch name.
+    Transforms targets into dist-git branch names.
 
     Args:
-        name: Name(s) of the system and version (e.g. "fedora-stable"
-            or "fedora-30") or branch name (e.g. "f30" or "epel8").
-        default: Used if no positional argument was given.
-        default_dg_branch: Default branch of dist-git repository.
-        with_aliases: If set to `True`, returns branches including aliases.
-            Can be used for reacting to webhooks where push can be done against
-            either the branch itself or its alias.
-
+        targets: List of targets, e.g. fedora-stable or epel-9.
+        default: Default to use if no target was specified.
+        default_dg_branch: Default branch of the dist-git repo.
+        with_aliases: Whether to include branch aliases, e.g. rawhide for main.
             Defaults to `False`.
 
     Returns:
         Set of dist-git branch names.
     """
-    if not (default or name):
-        return set()
-
-    names = list(name) or [default]
-    branches = set()
-
-    for sys_and_version in get_versions(*names):
-        if "rawhide" in sys_and_version:
-            branches.add(default_dg_branch)
-            if with_aliases:
-                branches.add("rawhide")
-        elif sys_and_version.startswith("fedora"):
-            sys, version = sys_and_version.rsplit("-", maxsplit=1)
-            branches.add(f"f{version}")
-        elif sys_and_version.startswith("epel"):
-            split = sys_and_version.rsplit("-", maxsplit=1)
-            if len(split) < 2:
-                branches.add(sys_and_version)
-                continue
-            sys, version = sys_and_version.rsplit("-", maxsplit=1)
-            if version.isnumeric() and int(version) <= 6:
-                branches.add(f"el{version}")
-            else:
-                branches.add(f"epel{version}")
-        else:
-            # We don't know, let's leave the original name.
-            branches.add(sys_and_version)
 
+    def branch_name(x):
+        if isinstance(x, Distro):
+            if "rawhide" in x.branch:
```

Comment:
```suggestion
            if x.branch == "rawhide":
```

Reply:
Hm, in the original code this was supposed to handle also `fedora-rawhide`, but in this implementation that would be converted to `Distro(namever="fedora-rawhide", branch="rawhide")`, so `x.branch == "rawhide"` should be correct. Unless the aliases fail to resolve for some reason, but that would break a lot of other things, so handling such case probably doesn't make sense.